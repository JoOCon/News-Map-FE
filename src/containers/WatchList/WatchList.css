.watch-list {
  display: flex;
  background: radial-gradient(#868686, #52535a);
  height: 100vh;
  width: 100%;
  overflow-y: auto; }

.favorites-list {
  width: 19rem;
  font-size: 0.6rem;
  border-right: 3px solid darkgray;
  background-color: #aaaaaa;
  overflow: auto; }
