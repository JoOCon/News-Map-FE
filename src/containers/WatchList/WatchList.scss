.watch-list {
  display: flex;
  flex-direction: column;
  padding: 1em 0 1em 1em;
  background: rgba(154, 154, 154, 0.9);
  border-radius: .5em;
}
