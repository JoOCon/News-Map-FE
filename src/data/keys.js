export const googleClientId =
  '241779944002-qa42v6cvk3kpu8da0jimdr364vn41t1r.apps.googleusercontent.com';
