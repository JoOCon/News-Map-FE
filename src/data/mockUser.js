export const mockUser = {
  El: '117977763637801404146',
  Zi: {
    token_type: 'Bearer',
    access_token:
      'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
    scope:
      'openid email profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/plus.me https://www.googleapis.com/auth/userinfo.profile',
    login_hint:
      'AJDLj6JUa8yxXrhHdWRHIV0S13cArCX2skrQ7skgtwzDVxi66KXli5pV9QkfHh_W71S4j789mYsJSJTDHdBzZgGY2XexHHgGyA',
    expires_in: 3600,
    id_token:
      'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
    session_state: { extraQueryParams: { authuser: '0' } },
    first_issued_at: 1540508499764,
    expires_at: 1540512099764,
    idpId: 'google'
  },
  w3: {
    Eea: '117977763637801404146',
    ig: 'Paul K',
    ofa: 'Paul',
    wea: 'K',
    Paa:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/ABtNlbDO0B2Sp9SzOXeqgK19boqsTiIVAA/s96-c/photo.jpg',
    U3: 'pykim0591@gmail.com'
  },
  googleId: '117977763637801404146',
  tokenObj: {
    token_type: 'Bearer',
    access_token:
      'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
    scope:
      'openid email profile https://www.googleapis.com/auth/userinfo.email https://www.googleapis.com/auth/plus.me https://www.googleapis.com/auth/userinfo.profile',
    login_hint:
      'AJDLj6JUa8yxXrhHdWRHIV0S13cArCX2skrQ7skgtwzDVxi66KXli5pV9QkfHh_W71S4j789mYsJSJTDHdBzZgGY2XexHHgGyA',
    expires_in: 3600,
    id_token:
      'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
    session_state: { extraQueryParams: { authuser: '0' } },
    first_issued_at: 1540508499764,
    expires_at: 1540512099764,
    idpId: 'google'
  },
  tokenId:
    'eyJhbGciOiJSUzI1NiIsImtpZCI6IjcyOGY0MDE2NjUyMDc5YjllZDk5ODYxYmIwOWJhZmM1YTQ1YmFhODYiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJhY2NvdW50cy5nb29nbGUuY29tIiwiYXpwIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwiYXVkIjoiMjQxNzc5OTQ0MDAyLXFhNDJ2NmN2azNrcHU4ZGEwamltZHIzNjR2bjQxdDFyLmFwcHMuZ29vZ2xldXNlcmNvbnRlbnQuY29tIiwic3ViIjoiMTE3OTc3NzYzNjM3ODAxNDA0MTQ2IiwiZW1haWwiOiJweWtpbTA1OTFAZ21haWwuY29tIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImF0X2hhc2giOiJNVS16RFc2c3lqbEhTNDZsbW5vMUVnIiwibmFtZSI6IlBhdWwgSyIsInBpY3R1cmUiOiJodHRwczovL2xoNC5nb29nbGV1c2VyY29udGVudC5jb20vLXE3OWJ2NE5jX2RZL0FBQUFBQUFBQUFJL0FBQUFBQUFBQUFBL0FCdE5sYkRPMEIyU3A5U3pPWGVxZ0sxOWJvcXNUaUlWQUEvczk2LWMvcGhvdG8uanBnIiwiZ2l2ZW5fbmFtZSI6IlBhdWwiLCJmYW1pbHlfbmFtZSI6IksiLCJsb2NhbGUiOiJlbiIsImlhdCI6MTU0MDUwODUwMCwiZXhwIjoxNTQwNTEyMTAwLCJqdGkiOiI2ZTYyNTg0Mzc2NTYwNDMwNjA1Y2MxYmVlNTUxOTE5ZmNjNTVmN2VmIn0.HFfyTNUGKeLor17A-qWqSRNxfITUysVYoPwC2UFzbn06Ulx4GEj7NDm3qMXFQCxXO50dGvOcsF8MKvB2fsmRkdVZeoin5QCdZa43xbfxoTmUCPX-zKFNUt8Ln3rcewNuQubkZ15Tq1BvaIh-vOQLxGSi7_yZXhTnnGLz1QNYaIZkdMUILJkGlOqigCN68GgumUWePIWn73Z-OGeKV0AStyLWpXRYtO6gIv1o8H_XV1WT2Iud-sTINDKWSK9iuomoFEnAHRcxbOQDX9A9yHHv7ZxIyzzKTd1QQmxnB_Aiux0mxu_zsZiZU5J9LS24QV5_EaAgWVKj3nCLYytEOGbjJQ',
  accessToken:
    'ya29.GlxBBtt0Yi5X7awZrvhwo7UAQjrOLQ_JwGQ0ZVcJhahzVb4kj_Z5NqMx4wAhouEXxtNs72jp9pG6TZ68H1nf3bgknrUoiWzhRnotssJFcYMiopKmF1oc0ejm8ZDCwQ',
  profileObj: {
    googleId: '117977763637801404146',
    imageUrl:
      'https://lh4.googleusercontent.com/-q79bv4Nc_dY/AAAAAAAAAAI/AAAAAAAAAAA/ABtNlbDO0B2Sp9SzOXeqgK19boqsTiIVAA/s96-c/photo.jpg',
    email: 'pykim0591@gmail.com',
    name: 'Paul K',
    givenName: 'Paul',
    familyName: 'K'
  }
};

export const mockStateUser = {
  first_name: 'Paul',
  last_name: 'K',
  email: 'pykim0591@gmail.com',
  gid: '117977763637801404146',
  id: 3,
  created_at: '2018-10-24T23:37:44.724Z',
  updated_at: '2018-10-24T23:37:44.724Z'
};

export const mockUserResponse = {
  created_at: '2018-10-24T23:37:44.724Z',
  email: 'pykim0591@gmail.com',
  first_name: 'Paul',
  gid: '117977763637801404146',
  id: 3,
  last_name: 'K',
  updated_at: '2018-10-24T23:37:44.724Z'
};
