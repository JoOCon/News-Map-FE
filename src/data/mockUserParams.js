export const responseGoogle1 =
  'https://event-mapper-api.herokuapp.com/api/v1/users';

export const responseGoogle2 = {
  body:
    '{"user":{"google_id":"117977763637801404146","given_name":"Paul","family_name":"K","email":"pykim0591@gmail.com"}}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const getEvents =
  'https://app.ticketmaster.com/discovery/v2/events.json?apikey=GddBLHQGZLJjPNWsNxOrYRAoBx8Hdqrf&geoPoint=9wjr4et3f&radius=200&size=100';

export const postUser = {
  body: '{"user":{"email":"pykim0591@gmail.com"}}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const setFavorite = {
  body: '{}',
  headers: { 'Content-Type': 'application/json' },
  method: 'POST'
};

export const removeEvent =
  'https://event-mapper-api.herokuapp.com/api/v1/users/undefined/events/undefined';

export const removeEventOptions = {
  headers: { 'Content-Type': 'application/json' },
  method: 'DELETE'
};
