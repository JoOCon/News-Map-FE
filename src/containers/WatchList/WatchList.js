import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { WatchListCard } from '../../components/WatchListCard/WatchListCard';
import { SelectedInfoContainer } from '../../components/SelectedInfoContainer/SelectedInfoContainer';
import * as call from '../../utilities/apiCalls/apiCalls';
import './WatchList.css';

export class WatchList extends Component {
  constructor() {
    super();
    this.state = {
      userWatchList: [],
      displayInfo: {}
      currentItem: null
    };
  }

  render() {
    const displayFavorites = this.state.userWatchList.map(favorite => (
      <WatchListCard handleSelection={this.handleSelection} key={favorite.e_id} {...favorite} />
    ));
    return (
      <div className='watch-list'>
        <div className='favorites-list'>
          {displayFavorites}
        </div>
        <SelectedInfoContainer displayInfo={this.state.displayInfo} />

  handleSelection = async (event, selectedItem) => {
    event.preventDefault();
    const { displayInfo } = this.state;
  
    if (currentItem !== selectedItem.id) {
      await call.getEventWeather(
        selectedItem.lat,
        selectedItem.lng,
        selectedItem.unix
      );
      this.setState({ selectedItem, currentItem: selectedItem.id });
    } else {
      this.setState({ currentItem: null });
    }
  };

  removeEvent = async (e, event) => {
    e.preventDefault();
    const { userWatchList } = this.state;
    const response = await removeFromWatchlist(
      this.props.activeUser.id,
      event.id
    );
    const userList = userWatchList.filter(item => item.e_id !== event.e_id);
    console.log(userWatchList);
    this.setState({ userWatchList: userList, currentItem: null });
  };

  render() {
    const { displayInfo, userWatchList, currentItem } = this.state;
    const displayFavorites = userWatchList.map(item => (
      <WatchListCard
        handleSelection={this.handleSelection}
        key={item.e_id}
        item={item}
      />
    ));
    return (
      <div className="watch-list">
        <div className='favorites-list'>
          {displayFavorites}
        </div>
        {currentItem && (
          <SelectedInfoContainer
            removeEvent={this.removeEvent}
            displayInfo={displayInfo}
          />
        )}
      </div>
    );
  }
}

WatchList.propTypes = {
  activeUser: PropTypes.object
};

export const mapStateToProps = state => ({
  activeUser: state.activeUser,
  watchlist: state.watchlist
});

export default connect(mapStateToProps)(WatchList);
