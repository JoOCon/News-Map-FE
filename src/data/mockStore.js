import * as user from './mockUser';
import * as events from './mockEvents';

export const mockStore = {
  activeUser: user.mockStateUser,
  userLocation: { latitude: 39.43243, longitude: -104.98434 },
  watchList: events.userWatchlist,
  events: events.eventsResponse
};
