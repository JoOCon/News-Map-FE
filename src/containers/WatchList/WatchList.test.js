import React from 'react';

import { WatchList } from './WatchList';
import { mockStateUser } from '../../data/mockUser';
import { eventStoreResponse } from '../../data/mockEvents';

jest.mock('../../utilities/apiCalls/apiCalls');

describe('WatchList', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <WatchList activeUser={mockStateUser} watchlist={eventStoreResponse} />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    wrapper.setState({ currentItem: 1, weather: 'sunny' });
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should setState of userWatchList on CDM', async () => {
    await wrapper.instance().getUserWatchlist();

    expect(wrapper.state().userWatchList).toEqual();
  });

  it.skip('should setState of displayInfo and currentItem if selected Item matches the currentItem state', () => {
    wrapper.setState({ currentItem: 1 });

    wrapper.instance().handleSelection({ id: 2 });

    expect(wrapper.state().displayInfo).toEqual({ id: 2 });
  });
});
