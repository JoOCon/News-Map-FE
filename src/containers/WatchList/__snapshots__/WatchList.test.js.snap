// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatchList should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WatchList
    activeUser={
      Object {
        "created_at": "2018-10-24T23:37:44.724Z",
        "email": "pykim0591@gmail.com",
        "first_name": "Paul",
        "gid": "117977763637801404146",
        "id": 3,
        "last_name": "K",
        "updated_at": "2018-10-24T23:37:44.724Z",
      }
    }
    watchlist={
      Array [
        Object {
          "address": "1000 Chopper Circle Denver, CO 80204",
          "date": "Mon, Apr 1, 2019 7:30 PM",
          "distance": 0.75,
          "e_id": "G5vzZ4amDV3z0",
          "favorite": false,
          "img": "https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg",
          "lat": "39.74724",
          "lng": "-105.010166",
          "name": "P!NK: BEAUTIFUL TRAUMA WORLD TOUR",
          "url": "https://www.altitudetickets.com/events/detail/pink-2019",
          "venue_name": "Pepsi Center",
        },
        Object {
          "address": "1000 Chopper Circle Denver, CO 80204",
          "date": "Tue, Dec 4, 2018 7:30 PM",
          "distance": 0.75,
          "e_id": "G5vzZ4aV5O3CO",
          "favorite": false,
          "img": "https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg",
          "lat": "39.74724",
          "lng": "-105.010166",
          "name": "Childish Gambino",
          "url": "https://www.altitudetickets.com/events/detail/childish-gambino",
          "venue_name": "Pepsi Center",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="favorites-list"
        />,
        null,
      ],
      "className": "watch-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "favorites-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="favorites-list"
          />,
          null,
        ],
        "className": "watch-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "favorites-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`WatchList should match snapshot 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WatchList
    activeUser={
      Object {
        "created_at": "2018-10-24T23:37:44.724Z",
        "email": "pykim0591@gmail.com",
        "first_name": "Paul",
        "gid": "117977763637801404146",
        "id": 3,
        "last_name": "K",
        "updated_at": "2018-10-24T23:37:44.724Z",
      }
    }
    watchlist={
      Array [
        Object {
          "address": "1000 Chopper Circle Denver, CO 80204",
          "date": "Mon, Apr 1, 2019 7:30 PM",
          "distance": 0.75,
          "e_id": "G5vzZ4amDV3z0",
          "favorite": false,
          "img": "https://s1.ticketm.net/dam/a/1dd/d5e86d93-5e1a-49d9-b530-70fefc0f21dd_877061_RETINA_PORTRAIT_3_2.jpg",
          "lat": "39.74724",
          "lng": "-105.010166",
          "name": "P!NK: BEAUTIFUL TRAUMA WORLD TOUR",
          "url": "https://www.altitudetickets.com/events/detail/pink-2019",
          "venue_name": "Pepsi Center",
        },
        Object {
          "address": "1000 Chopper Circle Denver, CO 80204",
          "date": "Tue, Dec 4, 2018 7:30 PM",
          "distance": 0.75,
          "e_id": "G5vzZ4aV5O3CO",
          "favorite": false,
          "img": "https://s1.ticketm.net/dam/a/7d9/44a96192-d246-4946-b6b4-c1f7f975a7d9_816871_TABLET_LANDSCAPE_3_2.jpg",
          "lat": "39.74724",
          "lng": "-105.010166",
          "name": "Childish Gambino",
          "url": "https://www.altitudetickets.com/events/detail/childish-gambino",
          "venue_name": "Pepsi Center",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="favorites-list"
        />,
        null,
      ],
      "className": "watch-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "favorites-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="favorites-list"
          />,
          null,
        ],
        "className": "watch-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "favorites-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
